File: src/main/java/org/ubdev/task/_3227/solution.rs
impl Solution {
    pub fn does_alice_win_my(s: String) -> bool {
        let res: String = s.chars().filter(|&x| "aeiou".contains(x)).collect();
        if res.len() > 0 {
            true
        } else {
            false
        }
    }

    pub fn does_alice_win(s: String) -> bool {
        let res: String = s.chars().filter(|&x| "aeiou".contains(x)).collect();
        if res.len() > 0 {
            true
        } else {
            false
        }
    }
}

File: src/main/java/org/ubdev/task/_3227/solution.cpp
#include <algorithm>
#include <string>

using namespace std;

class Solution {
public:
  bool doesAliceWin_my(string s) {
    for (auto &c : s) {
      switch (c) {
      case 'a':
      case 'e':
      case 'i':
      case 'o':
      case 'u':
        return true;
      }
    }
    return false;
  }

  static bool doesAliceWin(string &s) {
    // 0x104111 = 0b100000100000100010001
    return any_of(s.begin(), s.end(),
                  [](char c) { return ((0x104111 >> (c - 'a')) & 1); });
  }
};
